src/common/string/compare.ts

```diff
@@ -9,5 +9,26 @@ export const stringCompare = (a: string, b: string) => {
   return 0;
 };
 
-export const caseInsensitiveStringCompare = (a: string, b: string) =>
-  stringCompare(a.toLowerCase(), b.toLowerCase());
+export const stringCompare = (
+  a: string,
+  b: string,
+  language: string | undefined = undefined
+) => {
+  try {
+    return new Intl.Collator(language).compare(a, b);
```

Comment:
We should not create a new `Collator` for every compare, use a memoize so we cache it